gap/matrix/slconstr.gi

```diff
@@ -2068,126 +2068,127 @@ end;
-        if r = fail then
-           j := j+1;
-        else
-           i := 1;
-           t:= r^(p^(e*(d-2)) -1);
-           repeat
-              u1 := t^PseudoRandom(bbg);
-              if (not (Comm(t,u1)^p=One(bbg))) then
-                 u2 := t^PseudoRandom(bbg);
-                 sl3 := SubgroupNC(bbg,[t,u1,u2]);
-                 data1 := SLCR.QuickSL3DataStructure( sl3, p, e );
-                 if not data1 = fail then
-Print("constructed L with j=",j,"   i=",i,"  limit=",Int(2*(1- 1/q)^(-5)),"\n");
-                    stop := true;
-                 fi;
-              else
-                 i := i+1;
-              fi;
-           until stop or i > 2 * (1- 1/q)^(-5);
-           if not stop then
-              j := j+1;
-           else
-              genrec := SLCR.SLFindGenerators(bbg,data1,r,p,e,d);
-              if genrec = fail then
-                 stop := false;
-                 j := j+1;
-              fi;
-           fi;
-        fi; # r = fail
-      until stop or j=9;
-      if not stop then
-Print("could not construct L \n");
-return fail;
-      fi;
-   else # d>4 or q is not one of the bad values
-      # now r is guaranteed to be a transvection, if bbg is really SL(d,q)
-      if d>4 then
-         r := SLCR.FindGoodElement( bbg , p , e , d , 4 * q * (d-2) * Log(d,2) );
-      else
-         r := SLCR.SL4FindGoodElement( bbg , p , e , 16 * q );
-      fi;
-      if r = fail then
-Print("SLCR.FindGoodElement could not find r \n");
-return fail;
-      fi;
-      t:= r^(p^(e*(d-2)) -1);
-      stop := false;
-      i := 1;
-      repeat
-        u1 := t^PseudoRandom(bbg);
-        if (not (Comm(t,u1)^p=One(bbg))) then
-              u2 := t^PseudoRandom(bbg);
-              sl3 := SubgroupNC(bbg,[t,u1,u2]);
-              data1 := SLCR.QuickSL3DataStructure( sl3, p, e );
-              if not data1 = fail then
-Print("constructed L with i=",i,"  limit=",Int((1- 1/q)^(-5)* Log(8*d^2,2)),"\n");
-                 stop := true;
-              fi;
-        else
-           i := i+1;
-        fi;
-      until stop or (i > (1- 1/q)^(-5) * Log(8*d^2,2));
-      if not stop then
-Print("could not construct L \n");
-return fail;
-      fi;
-      genrec := SLCR.SLFindGenerators(bbg,data1,r,p,e,d);
-      if genrec = fail then
-return fail;
-      fi;
-
-   fi; # d=4 and q in [2,3,5,9,17]
-
-   Lgen := genrec.Lgen;
-   Q := genrec.Q;
-   Qgamma := genrec.Qgamma;
-   pQ := genrec.pQ;
-
-   L := SubgroupNC(bbg,Lgen);
-
-   data2 := SLCR.SLDataStructure(L, p, e, d-1);
-   if data2 = fail then
-return fail;
-   fi;
-
-   vectors := SLCR.SLExchangeL(data2, Q, Qgamma, pQ, d);
-   if vectors = fail then
-return fail;
-   fi;
-   data3 := SLCR.SLConstructBasisQ(bbg, data2, vectors[1], vectors[2], d, q);
-   data2 := SLCR.SLFinishConstruction(bbg,data2,data3,d);
-return data2;
-end;
+#
+#SLCR.SLDataStructure := function( bbg, p, e, d )
+#    local i, j, stop, q, r, t, u1, u2, sl3, data1, Q, pQ, Lgen,
```

Comment:
I am not a fan of commenting out (*and* reformatting, at the same time...) this code. Alice and me want to work on it soon (assuming our grant proposal is accepted0

Reply:
Ok, I'll drop the commit.